### PR TITLE
fix(oracle): Above price not removed

### PR DESCRIPTION
**Pull request by KirilMihaylov**

### CODE REVIEW

File: tests/src/oracle_tests.rs [hunk 1]
```diff
@@ -188,6 +188,67 @@ fn feed_price_with_alarm() {
...
+            lease,
+            test_case.oracle.clone().unwrap(),
+            &oracle::msg::ExecuteMsg::DispatchAlarms { max_count: 5 },
+            &[],
+        )
+        .unwrap();
+
+    dbg!(res);
+}
+
```
Comment (gmanev7) [inline tests/src/oracle_tests.rs:251]: no assert-s?? what test is it like? Please add the necessary ones

File: tests/src/oracle_tests.rs [hunk 2]
```diff
@@ -188,6 +188,79 @@ fn feed_price_with_alarm() {
...
+    );
+
+    platform::tests::assert_event(
+        &res.events,
+        &Event::new("reply").add_attribute("mode", "handle_success"),
+    );
+
+    platform::tests::assert_event(
+        &res.events,
+        &Event::new("wasm-market-alarm").add_attribute("delivered", "success"),
```
Comment (gmanev7) [inline tests/src/oracle_tests.rs:260]: a side question: "wasm-pricealarm" and "wasm-market-alarm" sound too similar. May we update them with more descriptive names?

Comment (KirilMihaylov) [reply tests/src/oracle_tests.rs:260]: I am not sure how many of the other tests this change will affect so I'd rather leave it out of this PR.

File: contracts/oracle/src/contract/oracle/mod.rs [hunk 3]
```diff
@@ -129,6 +129,15 @@ where
...
 
+        #[cfg(debug_assertions)]
+        {
+            use std::collections::HashSet;
+
+            let set: HashSet<&Addr> = HashSet::from_iter(subscribers.as_slice());
+
+            debug_assert_eq!(set.len(), subscribers.len());
+        }
+
```
Comment (gmanev7) [inline contracts/oracle/src/contract/oracle/mod.rs:140]: please extract this chunk into a function ... we shall keep the function body succint

File: packages/marketprice/src/alarms/mod.rs [hunk 4]
```diff
@@ -205,7 +205,13 @@ where
         self.add_alarm_above_or_equal_internal(subscriber, &NormalizedPrice::new(&alarm))
     }
 
-    pub fn remove(&mut self, subscriber: Addr) -> Result<(), AlarmError> {
+    pub fn remove_above_or_equal(&mut self, subscriber: Addr) -> Result<(), AlarmError> {
+        self.alarms_above_or_equal
+            .remove(self.storage.deref_mut(), subscriber)
+            .map_err(Into::into)
+    }
```
Comment (gmanev7) [inline packages/marketprice/src/alarms/mod.rs:212]: let's avoid copy/pasting code by calling this from `fn remove_all`

**Review by gmanev7 — COMMENTED** [review]
just a few improvements needed

File: tests/src/oracle_tests.rs (hunk 2 above)
Comment (gmanev7) [reply tests/src/oracle_tests.rs:260]: agree, going to file a ticket

**Comment by gmanev7** [comment]
please bump the Cargo packages' versions

**Comment by gmanev7** [comment]
> please bump the Cargo packages' versions

please note the plural voice

**Review by gmanev7 — APPROVED** [review]
LGTM

**Pull request merged**